src/getJavaHome.ts

```diff
@@ -39,6 +39,14 @@ function fromConfig(
     typeof configuredJavaHome === "string" &&
     configuredJavaHome.trim() !== ""
   ) {
+    if (fs.existsSync(configuredJavaHome)) {
+      const stat = fs.lstatSync(configuredJavaHome);
+      const realpath = fs.realpathSync(configuredJavaHome);
+      if (stat.isSymbolicLink() && realpath.endsWith(`bin${path.sep}java`)) {
+        const javaHome = path.dirname(path.dirname(realpath));
+        return TE.right(javaHome);
+      }
+    }
```

Comment:
Is it more readable? I find flat ifs and early return more readable than nesting it
```typescript
  const cleanJavaHome = configuredJavaHome?.trim();

  if (cleanJavaHome == null) return TE.left({});

  if (cleanJavaHome.length === 0) return TE.left({});

  if (fs.existsSync(cleanJavaHome)) {
    const stat = fs.lstatSync(cleanJavaHome);
    const realpath = fs.realpathSync(cleanJavaHome);
    if (stat.isSymbolicLink() && realpath.endsWith(`bin${path.sep}java`)) {
      const javaHome = path.dirname(path.dirname(realpath));
      return TE.right(javaHome);
    }
  }
  
  return TE.right(cleanJavaHome);
```